### PR TITLE
(PCP-344) Add ttl_expired callback

### PR DESCRIPTION
**Pull request by parisiale**

### CODE REVIEW

File: lib/src/connector/connector.cc [hunk 1]
```diff
@@ -561,14 +574,45 @@ void Connector::errorMessageCallback(const ParsedChunks& parsed_chunks) {
...
+                expired_msg_id);
+
+    if (TTL_expired_callback_)
+        TTL_expired_callback_(parsed_chunks);
+
+    if (session_association_.in_progress.load()) {
+        Util::lock_guard<Util::mutex> the_lock { session_association_.mtx };
+
+        if (!expired_msg_id.empty()
+                && expired_msg_id == session_association_.request_id) {
```
Comment (mruzicka) [inline lib/src/connector/connector.cc]: Just wondering: Which component would be sending the TTL expired message in this instance?

**Comment by mruzicka** [comment]
LGTM

File: lib/src/connector/connector.cc [hunk 2]
```diff
@@ -561,14 +574,45 @@ void Connector::errorMessageCallback(const ParsedChunks& parsed_chunks) {
...
+    auto expired_msg_id = parsed_chunks.data.get<std::string>("id");
+
+    LOG_WARNING("Received TTL Expired message %1% from %2% related to message %3%",
+                ttl_msg_id, parsed_chunks.envelope.get<std::string>("sender"),
+                expired_msg_id);
+
+    if (TTL_expired_callback_)
+        TTL_expired_callback_(parsed_chunks);
+
+    if (session_association_.in_progress.load()) {
```
Comment (mruzicka) [inline lib/src/connector/connector.cc]: Should this go to a TTLExpiredCallback, which would only be registered until the session association is completed?

File: lib/src/connector/connector.cc (hunk 1 above)
Comment (parisiale) [inline lib/src/connector/connector.cc]: ttl_expired messages are only sent by the broker

File: lib/src/connector/connector.cc (hunk 2 above)
Comment (parisiale) [inline lib/src/connector/connector.cc]: No.

**Pull request merged**